OS/2 port of FeatherPad

josch1710 opened the pull request:
To have a lightweight editor for porting stuff, I also ported FeatherPad to OS/2.

tsujan commented:
Thanks!

I can't test it under OS/2. Does everything work fine? Hard-coded icons, saving of settings,... Spell checker too?

tsujan commented on featherpad/fpwin.cpp:
```diff
@@ -695,7 +695,7 @@ void FPwin::applyConfigOnStarting()
         config.readShortcuts();
     }
-
+    
```
Would you please remove this redundant change and also put the OS/2-specific file `featherpad.rc` inside the `data` directory?

josch1710 commented:
Hi.
I removed the tab and moved featherpad.rc to data. I also deactivated the -mtune flags for now, because something doesn't work out outside rpmbuild. I have to investigate it.

As for the port itself: Icons are there, settings are saved etc. The only thing, which doesn't activate at the moment, is the Run menu item. I still have to investigate that. 
[Screenshot](https://www.os2world.com/forum/index.php?action=dlattach;topic=2859.0;attach=7679;image)

tsujan commented:
Thanks for the changes!

I also had a question about the SVG icon. It's used with qmake but not with cmake. Is that intentional?

josch1710 commented:
No, I just finished the port using cmake, because that works better.
I have to look at the qmake stuff again for more parity.

tsujan commented:
> The only thing, which doesn't activate at the moment, is the Run menu item. I still have to investigate that.

Maybe, on OS/2, executable files cannot be executed only by entering their paths. FeatherPad has a setting for the command in Preferences → Files.

tsujan commented:
> I have to look at the qmake stuff again for more parity.

OK. Please tell me when your work is finished. I want to merge this.

josch1710 commented:
> > The only thing, which doesn't activate at the moment, is the Run menu item. I still have to investigate that.
> 
> Maybe, on OS/2, executable files cannot be executed only by entering their paths. FeatherPad has a setting for the command in Preferences → Files.

Thanks for the hint. I haven't explored all nooks and crannies of FeatherPad, yet ;-)
But otherwise, the port was a breeze.

josch1710 commented:
So, "Run" works with the setting. Thanks.

I reworked the qmake project and renamed featherpad.rc to featherpad_os2.rc to be clear, that the file is used for OS/2 and not Windows. Also it's more with featherpad_os2.ico.

I found two typos in the german translation file and fixed them.
I'm finished for now ;-)

tsujan commented:
Thank you very much!

I'll add a line about OS/2 to `README.md` later. Merging…

josch1710 commented:
I also have binaries in my release section.

tsujan commented:
> I also have binaries in my release section.

If FeatherPad's `INSTALL` file was as detailed as [that of Kvantum](https://github.com/tsujan/Kvantum/blob/master/Kvantum/INSTALL.md), I would add a link to it but `INSTALL` is simple and generic.

josch1710 commented:
It's fine. Just wanted to let you know.

tsujan commented:
@josch1710
Could you please check that the qmake compilation works with the latest git FeatherPad? — after this commit: https://github.com/tsujan/FeatherPad/commit/0132aeb95ce727feb3f3feb97ccb9355f84b46c0

josch1710 commented:
@tsujan 
I have made a pull request.
